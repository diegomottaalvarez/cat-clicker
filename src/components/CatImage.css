.cat-img {
  border-radius: 1rem;
  margin: 1rem 0;

  max-width: 30rem;
  max-height: 18rem;
  width: auto;
  height: auto;
}

@media (max-width: 900px) {
  .cat-img {
    border-radius: 1rem;
    margin: 1rem 0;

    max-width: 100%;
    max-height: 100%;
    width: auto;
    height: auto;
  }
}
