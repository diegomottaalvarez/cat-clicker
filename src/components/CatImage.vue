<template>
  <img
    v-if="catDetail.url"
    :src="catDetail.url"
    :alt="catDetail.breeds[0].name"
    class="cat-img"
  />
</template>

<script>
export default {
  props: {
    catDetail: Object,
  },
};
</script>

<style scoped>
@import './CatImage.css';
</style>
