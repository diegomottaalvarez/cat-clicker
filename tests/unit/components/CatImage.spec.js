import { shallowMount } from '@vue/test-utils';
import CatImage from '@/components/CatImage.vue';
import { catDetailMock } from '@/helpers/getCats';

describe('CatImage component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(CatImage, {
      propsData: {
        catDetail: catDetailMock,
      },
    });
  });
  test('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should paint cat image', () => {
    const img = wrapper.find('img');
    expect(img.exists()).toBeTruthy();
  });
});
