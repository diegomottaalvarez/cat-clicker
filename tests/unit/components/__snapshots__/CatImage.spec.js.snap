// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatImage component should match the snapshot 1`] = `<img src="https://cdn2.thecatapi.com/images/K4WzCAYWm.jpg" alt="Somali" class="cat-img">`;
